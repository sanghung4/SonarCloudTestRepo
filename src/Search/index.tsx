import React, { useContext, useEffect, useMemo, useState } from 'react';

import {
  Box,
  Button,
  Container,
  Divider,
  Grid,
  Hidden,
  Pagination,
  Typography,
  useScreenSize
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';
import { Redirect, useHistory } from 'react-router-dom';
import { kebabCase } from 'lodash-es';

import { AuthContext } from 'AuthProvider';
import BackToTop from 'common/BackToTop';
import {
  AggregationResults,
  Product,
  ProductSearchResult,
  useGetProductPricingQuery,
  useSearchProductQuery
} from 'generated/graphql';
import SearchBreadcrumbs from 'Search/SearchBreadcrumbs';
import SearchFilters from 'Search/SearchFilters';
import SearchResult from 'Search/SearchResult';
import SearchResultSkeleton from 'Search/SearchResultSkeleton';
import SearchNoResults from 'Search/SearchNoResults';
import SearchVisitorAlert from 'Search/SearchVisitorAlert';
import useSearchQueryParams from 'Search/util/useSearchQueryParams';
import { useDomainInfo } from 'hooks/useDomainInfo';
import { HeaderContext } from 'common/Header/HeaderProvider';
import { useSelectedAccountsContext } from 'providers/SelectedAccountsProvider';
import { BranchContext } from 'providers/BranchProvider';
import FeaturedSearch from 'Search/FeaturedSearch';
import AdvancedToolTip from 'common/AdvancedToolTip';
import { WarningIcon } from 'icons';
import { useCartContext } from 'providers/CartProvider';

/**
 * Types
 */
export type Filters = Pick<
  ProductSearchResult,
  'selectedAttributes' | 'selectedCategories' | 'categoryLevel'
>;

export type LocationStateSearch = {
  categories: string[];
  filters: ProductSearchResult['selectedAttributes'];
};

function Search() {
  /**
   * Custom components
   */
  const history = useHistory();
  const { isSmallScreen } = useScreenSize();
  const { t } = useTranslation();
  const { engine } = useDomainInfo();

  const [params, setParams] = useSearchQueryParams();
  const { criteria = '', page = 1, categories = [], filters = [] } = params;

  /**
   * Context
   */
  const { authState, activeFeatures, profile } = useContext(AuthContext);
  const { setSearchPage, setTrackedSearchTerm, setPageIndex } =
    useContext(HeaderContext);
  const { selectedAccounts } = useSelectedAccountsContext();
  const { shippingBranch } = useContext(BranchContext);
  const { disableAddToCart } = useCartContext();

  /**
   * State
   */
  const [filtersOpen, setFiltersOpen] = useState(false);
  const [totalCount, setTotalCount] = useState(0);

  /**
   * Memos
   */
  const filterCount = useMemo(filterCountMemo, [categories, filters]);
  const pageSize = activeFeatures?.includes('POST_SEARCH_API') ? 24 : 12;

  /**
   * Queries
   */
  const { data, previousData, loading } = useSearchProductQuery({
    fetchPolicy: 'no-cache',
    variables: {
      productSearch: {
        searchTerm: criteria.replace('"', '\\"'),
        page: Number(page),
        size: pageSize,
        categoryLevel: categories.length,
        categories,
        filters,
        engine,
        state: selectedAccounts.shipToErpAccount?.state?.toLowerCase(),
        customerNumber: selectedAccounts.billToErpAccount?.erpAccountId
      },
      userId: profile?.userId ?? '',
      shipToAccountId: selectedAccounts.shipTo?.id ?? ''
    },
    onCompleted: (data) => {
      setTotalCount(data.searchProduct?.pagination?.totalItemCount ?? 0);
    }
  });

  const partNumbers = useMemo(partNumbersMemo, [data?.searchProduct?.products]);

  const { data: pricingData, loading: pricingLoading } =
    useGetProductPricingQuery({
      fetchPolicy: 'no-cache',
      skip: Boolean(!authState?.isAuthenticated || !data?.searchProduct),
      variables: {
        input: {
          customerId: selectedAccounts.billTo?.erpAccountId ?? '',
          branchId: shippingBranch?.branchId ?? '',
          productIds: partNumbers,
          includeListData: true
        }
      }
    });

  /**
   * Effects
   */
  useEffect(handleBadPage, [history, page, params, setParams]);

  useEffect(() => {
    if (
      shippingBranch?.isPricingOnly &&
      filters.includes('inStockLocation|in_stock_location')
    ) {
      setParams({
        ...params,
        page: '1',
        filters: filters.filter(
          (filter) => filter !== 'inStockLocation|in_stock_location'
        )
      });
    }
  }, [shippingBranch?.isPricingOnly, filters, params, setParams]);

  /**
   * Callbacks
   */
  const handlePagination = (newPage: number) => {
    window.scrollTo({ top: 0, behavior: 'smooth' });
    setParams({
      ...params,
      page: newPage.toString()
    });
  };

  const PaginationElement = () => (
    <Pagination
      count={Math.ceil(
        (data?.searchProduct?.pagination?.totalItemCount || 0) / pageSize
      )}
      current={Number(page)}
      ofText={t('common.of')}
      onChange={handlePagination}
      onPrev={handlePagination}
      onNext={handlePagination}
      data-testid="pagination-container"
      dataTestIdCurrentPage="current-page-number"
      dataTestIdTotalNumberOfPages="total-number-of-pages"
    />
  );

  const searchString = (value: string) => {
    if (value) {
      const newSearchValue = value.replaceAll(/(")\1{1,}/g, '"');
      return `${newSearchValue} (${totalCount} ${
        totalCount !== 1 ? 'results' : 'result'
      })`;
    } else {
      return null;
    }
  };

  return criteria === null ? (
    <Redirect to="/" />
  ) : !loading &&
    (data?.searchProduct?.products?.length === 0 || !data?.searchProduct) ? (
    <SearchNoResults searchTerm={criteria} />
  ) : (
    <>
      <Hidden mdUp>
        <SearchFilters
          loading={Boolean(previousData) && loading}
          count={data?.searchProduct?.pagination?.totalItemCount || 0}
          filters={data?.searchProduct?.aggregates as AggregationResults}
          filtersOpen={filtersOpen}
          onShowHide={(value) => setFiltersOpen(value)}
        />
      </Hidden>
      <Box
        flex="1"
        pb={isSmallScreen ? 6 : 12}
        sx={{ bgcolor: 'common.white' }}
      >
        {!authState?.isAuthenticated && !loading ? (
          <SearchVisitorAlert />
        ) : null}
        <Hidden mdUp>
          <Box m={3}>
            <Grid container alignItems="center">
              <Grid item xs>
                {!loading ? <SearchBreadcrumbs /> : null}
              </Grid>
              <Grid
                container
                xs
                justifyContent="flex-end"
                alignItems="center"
                data-testid={kebabCase(
                  `${
                    categories.length !== 0
                      ? categories[categories.length - 1]
                      : criteria
                  }-title`
                )}
              >
                <Typography
                  component="span"
                  textAlign="right"
                  sx={{ fontWeight: 700 }}
                >
                  &nbsp;
                  {categories.length !== 0
                    ? categories[categories.length - 1]
                    : criteria}
                </Typography>
                <Typography
                  variant="subtitle1"
                  display="block"
                  sx={{ fontWeight: 700 }}
                >
                  &nbsp; ({totalCount} {totalCount !== 1 ? 'results' : 'result'}
                  )
                </Typography>
              </Grid>
            </Grid>
          </Box>
        </Hidden>
        <Container
          maxWidth="lg"
          sx={(theme) => ({
            [theme.breakpoints.down('md')]: {
              py: 0,
              px: 4
            }
          })}
        >
          <Grid container spacing={6} sx={{ mt: 0 }}>
            <Hidden mdDown>
              <Grid item md={3}>
                <Box
                  mt={0}
                  display="flex"
                  flexDirection="column"
                  alignItems="flex-end"
                >
                  <SearchFilters
                    loading={loading}
                    count={data?.searchProduct?.pagination?.totalItemCount || 0}
                    filters={
                      data?.searchProduct?.aggregates as AggregationResults
                    }
                    filtersOpen={filtersOpen}
                    onShowHide={(value: boolean) => setFiltersOpen(value)}
                  />
                </Box>
              </Grid>
            </Hidden>
            <Grid item xs={12} md={9}>
              {activeFeatures?.includes('FEATURED_SEARCH') && (
                <FeaturedSearch />
              )}
              <Hidden mdDown>
                {!loading ? <SearchBreadcrumbs /> : null}
                <Box mt={2} mb={5}>
                  <Grid container alignItems="center">
                    <Grid
                      item
                      md={6}
                      data-testid={
                        categories.length
                          ? kebabCase(
                              `${categories[categories.length - 1]}-title`
                            )
                          : kebabCase(`${criteria}-title`)
                      }
                    >
                      <Typography
                        variant="h5"
                        component="span"
                        sx={{ fontWeight: 700 }}
                      >
                        {criteria
                          ? searchString(criteria)
                          : searchString(categories[categories.length - 1])}
                      </Typography>
                    </Grid>
                    <Grid
                      container
                      item
                      md={6}
                      justifyContent="flex-end"
                      data-testid="pagination-counter-top"
                    >
                      {/* {!loading ? <SearchSort /> : null} */}
                      {!loading &&
                      (data?.searchProduct?.pagination?.totalItemCount || 0) >
                        0 ? (
                        <PaginationElement />
                      ) : null}
                    </Grid>
                  </Grid>
                </Box>
              </Hidden>
              <Grid container spacing={2} data-testid="search-container">
                {loading ? (
                  [...new Array(3)].map((_, i) => (
                    <Grid
                      container
                      item
                      md={3}
                      direction="column"
                      key={i}
                      sx={(theme) => ({
                        '&:not(:last-child)': {
                          [theme.breakpoints.down('md')]: {
                            mb: 4
                          }
                        }
                      })}
                    >
                      <SearchResultSkeleton
                        testid={`search-result-skeleton-${i}`}
                      />
                    </Grid>
                  ))
                ) : (
                  <>
                    <Hidden mdUp>
                      <Grid
                        item
                        xs
                        mb={disableAddToCart && isSmallScreen ? 16 : undefined}
                      >
                        <AdvancedToolTip
                          title="Warning"
                          text={t('cart.maxLimitToolTip')}
                          icon={<WarningIcon />}
                          placement="bottom"
                          disabled={disableAddToCart && isSmallScreen}
                        >
                          <Button
                            variant="secondary"
                            size="small"
                            fullWidth
                            onClick={() => setFiltersOpen(true)}
                            sx={{ mb: 3 }}
                            data-testid="filter-count-mob"
                          >
                            {t('common.filter')}
                            {filterCount > 0 ? ` (${filterCount})` : null}
                          </Button>
                        </AdvancedToolTip>
                      </Grid>
                    </Hidden>
                    {data?.searchProduct?.products?.map((product, i) => (
                      <Grid
                        container
                        item
                        md={3}
                        direction="column"
                        key={i}
                        sx={(theme) => ({
                          '&:not(:last-child)': {
                            [theme.breakpoints.down('md')]: {
                              mb: 4
                            }
                          }
                        })}
                      >
                        <SearchResult
                          product={product as Product}
                          pricing={pricingData?.productPricing.products ?? []}
                          pricingLoading={pricingLoading}
                          index={i}
                          onClick={() => handleResultClick(i)}
                        />
                        {isSmallScreen && (
                          <Box pt={3}>
                            <Divider />
                          </Box>
                        )}
                      </Grid>
                    ))}
                  </>
                )}
              </Grid>
              {!loading &&
              (data?.searchProduct?.pagination?.totalItemCount || 0) > 0 ? (
                <Box
                  mt={isSmallScreen ? 6 : 4}
                  mb={isSmallScreen ? 6 : 0}
                  display="flex"
                  justifyContent="center"
                  data-testid="pagination-counter-bottom"
                >
                  <PaginationElement />
                </Box>
              ) : null}
            </Grid>
          </Grid>
        </Container>
      </Box>
      {!loading && isSmallScreen ? <BackToTop /> : null}
    </>
  );

  /**
   * Memo defs
   */
  function filterCountMemo() {
    return categories.length + filters.length;
  }

  function partNumbersMemo() {
    return (data?.searchProduct?.products || []).map(
      (product) => product.partNumber ?? ''
    );
  }

  /**
   * Effect defs
   */
  function handleBadPage() {
    if (page !== null && page < 0) {
      setParams({ ...params, page: '1' });
    }
  }

  /**
   * Callback Definitions
   */
  function handleResultClick(pageIndex: number) {
    setTrackedSearchTerm(criteria);
    setSearchPage(Number(page));
    setPageIndex(pageIndex);
  }
}

export default Search;
