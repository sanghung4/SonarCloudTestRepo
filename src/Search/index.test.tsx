import { waitFor } from '@testing-library/react';

import Search from 'Search';
import mocks, { mockNoProducts } from 'Search/index.mocks';
import { SelectedAccounts } from 'providers/SelectedAccountsProvider';
import { render } from 'test-utils/TestWrapper';
import { setBreakpoint } from 'test-utils/mockMediaQuery';

const selectedAccountsMock: SelectedAccounts = {
  shipTo: {
    id: 'shipto'
  }
};

jest.mock('Search/util/useSearchQueryParams', () => ({
  ...jest.requireActual('Search/util/useSearchQueryParams'),
  __esModule: true,
  default: () => [
    {
      criteria: 'copper',
      categories: [],
      filters: [],
      page: 1
    },
    jest.fn()
  ]
}));

jest.mock('hooks/useDomainInfo', () => ({
  useDomainInfo: () => ({
    engine: 'plumbing_hvac'
  })
}));

describe('Search tests', () => {
  beforeAll(() => setBreakpoint());

  it('Should make separate calls for the product and the pricing & availability', async () => {
    const { getByTestId } = render(<Search />, {
      mocks
    });

    const searchResultContainer = getByTestId('search-container');
    const searchResultSkeleton = getByTestId('search-result-skeleton-0');
    expect(searchResultContainer).toMatchSnapshot('1 > loading');
    expect(searchResultSkeleton).toBeInTheDocument();

    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    expect(searchResultContainer).toMatchSnapshot('2 > price loading');
    expect(searchResultSkeleton).not.toBeInTheDocument();

    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    expect(searchResultContainer).toMatchSnapshot('3 > loaded');
  });

  it('show search results for the given criteria', async () => {
    const { container, getByTestId } = render(<Search />, {
      selectedAccountsConfig: { selectedAccounts: selectedAccountsMock },
      mocks
    });

    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    const zeroResultsMsg = getByTestId('search-container');

    expect(zeroResultsMsg).toBeInTheDocument();
  });

  it('empty products show show the zero results message', async () => {
    const { getByTestId } = render(<Search />, {
      selectedAccountsConfig: { selectedAccounts: selectedAccountsMock },
      mocks: mockNoProducts
    });

    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    const zeroResultsMsg = getByTestId('zero-results-message');

    expect(zeroResultsMsg).toBeInTheDocument();
  });
});
