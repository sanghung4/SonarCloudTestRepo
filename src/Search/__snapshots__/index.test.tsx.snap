// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search tests Should make separate calls for the product and the pricing & availability: 1 > loading 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  data-testid="search-container"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-md-3 css-vis2b6-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-19midj6"
      data-testid="search-result-skeleton-0"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1pfza9f-MuiSkeleton-root"
        style="width: 100%; height: 120px;"
      />
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
          style="height: 3.5rem;"
        />
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <div
        class="MuiBox-root css-1yd7q65"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3x7rqn-MuiSkeleton-root"
          style="width: 45px; height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-66z2mb-MuiSkeleton-root"
          style="width: 100%; height: 30px;"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-md-3 css-vis2b6-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-19midj6"
      data-testid="search-result-skeleton-1"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1pfza9f-MuiSkeleton-root"
        style="width: 100%; height: 120px;"
      />
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
          style="height: 3.5rem;"
        />
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <div
        class="MuiBox-root css-1yd7q65"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3x7rqn-MuiSkeleton-root"
          style="width: 45px; height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-66z2mb-MuiSkeleton-root"
          style="width: 100%; height: 30px;"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-md-3 css-vis2b6-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-19midj6"
      data-testid="search-result-skeleton-2"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1pfza9f-MuiSkeleton-root"
        style="width: 100%; height: 120px;"
      />
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
          style="height: 3.5rem;"
        />
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <div
        class="MuiBox-root css-1yd7q65"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3x7rqn-MuiSkeleton-root"
          style="width: 45px; height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-66z2mb-MuiSkeleton-root"
          style="width: 100%; height: 30px;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Search tests Should make separate calls for the product and the pricing & availability: 2 > price loading 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  data-testid="search-container"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-md-3 css-vis2b6-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-19midj6"
      data-testid="search-result-skeleton-0"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1pfza9f-MuiSkeleton-root"
        style="width: 100%; height: 120px;"
      />
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
          style="height: 3.5rem;"
        />
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <div
        class="MuiBox-root css-1yd7q65"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3x7rqn-MuiSkeleton-root"
          style="width: 45px; height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-66z2mb-MuiSkeleton-root"
          style="width: 100%; height: 30px;"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-md-3 css-vis2b6-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-19midj6"
      data-testid="search-result-skeleton-1"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1pfza9f-MuiSkeleton-root"
        style="width: 100%; height: 120px;"
      />
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
          style="height: 3.5rem;"
        />
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <div
        class="MuiBox-root css-1yd7q65"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3x7rqn-MuiSkeleton-root"
          style="width: 45px; height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-66z2mb-MuiSkeleton-root"
          style="width: 100%; height: 30px;"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-md-3 css-vis2b6-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-19midj6"
      data-testid="search-result-skeleton-2"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1pfza9f-MuiSkeleton-root"
        style="width: 100%; height: 120px;"
      />
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
          style="height: 3.5rem;"
        />
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <div
        class="MuiBox-root css-1yd7q65"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3x7rqn-MuiSkeleton-root"
          style="width: 45px; height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-66z2mb-MuiSkeleton-root"
          style="width: 100%; height: 30px;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Search tests Should make separate calls for the product and the pricing & availability: 3 > loaded 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  data-testid="search-container"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-md-3 css-vis2b6-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-19midj6"
      data-testid="search-result-skeleton-0"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1pfza9f-MuiSkeleton-root"
        style="width: 100%; height: 120px;"
      />
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
          style="height: 3.5rem;"
        />
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <div
        class="MuiBox-root css-1yd7q65"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3x7rqn-MuiSkeleton-root"
          style="width: 45px; height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-66z2mb-MuiSkeleton-root"
          style="width: 100%; height: 30px;"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-md-3 css-vis2b6-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-19midj6"
      data-testid="search-result-skeleton-1"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1pfza9f-MuiSkeleton-root"
        style="width: 100%; height: 120px;"
      />
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
          style="height: 3.5rem;"
        />
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <div
        class="MuiBox-root css-1yd7q65"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3x7rqn-MuiSkeleton-root"
          style="width: 45px; height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-66z2mb-MuiSkeleton-root"
          style="width: 100%; height: 30px;"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-md-3 css-vis2b6-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-19midj6"
      data-testid="search-result-skeleton-2"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1pfza9f-MuiSkeleton-root"
        style="width: 100%; height: 120px;"
      />
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
          style="height: 3.5rem;"
        />
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption css-12u2emh-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        />
      </span>
      <div
        class="MuiBox-root css-1yd7q65"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3x7rqn-MuiSkeleton-root"
          style="width: 45px; height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-66z2mb-MuiSkeleton-root"
          style="width: 100%; height: 30px;"
        />
      </div>
    </div>
  </div>
</div>
`;
